### DISCUSSION TITLE
AbpAutoValidateAntiforgeryTokenAttribute

### DISCUSSION THREAD

**Discussion by cgrorhan** [abpframework/abp#9211] [Q&A]
I uploaded aspboilerplate EF version 2.2 to 3.1 and in the startup.cs folder I added this code block ;
`services.AddControllersWithViews(
                options =>
                {
                    options.Filters.Add(new AbpAutoValidateAntiforgeryTokenAttribute());
                }
            ).AddNewtonsoftJson(options =>
            {
                options.SerializerSettings.ContractResolver = new AbpMvcContractResolver(IocManager.Instance)
                {
                    NamingStrategy = new CamelCaseNamingStrategy()
                };
            });`

but the compiler can not find the AbpAutoValidateAntiforgeryTokenAttribute() , which package or version do I need to install?

**Comment by maliming** [comment]
Are you talking about [aspnetboilerplate](https://github.com/aspnetboilerplate/aspnetboilerplate)?

**Reply by cgrorhan** [reply]
Yes I figure it out that problem but that came out then ;
'Unhandled exception. System.TypeLoadException: Method 'IsGranted' in type 'Abp.Authorization.PermissionChecker`2' from assembly 'Abp.ZeroCore, Version=4.9.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.'

**Reply by maliming** [reply]
The version of abp packages in your project should be the same.
